src/cdk/explorer/awsCdkExplorer.ts

```diff
@@ -0,0 +1,73 @@
+/*!
+ * Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as fs from 'fs'
+import * as vscode from 'vscode'
+import { registerCommand } from '../../shared/telemetry/telemetryUtils'
+import { RefreshableAwsTreeProvider } from '../../shared/treeview/awsTreeProvider'
+import { AWSTreeNodeBase } from '../../shared/treeview/nodes/awsTreeNodeBase'
+import { CdkProject, detectCdkProjects } from './detectCdkProjects'
+import { ConstructNode } from './nodes/constructNode'
+import { ConstructTree, ConstructTreeEntity } from './tree/types'
+
+/**
+ * Provides data for the AWS CDK Explorer view
+ */
+export class AwsCdkExplorer implements vscode.TreeDataProvider<AWSTreeNodeBase>, RefreshableAwsTreeProvider {
+    public viewProviderId: string = 'aws.cdk.explorer'
+    public readonly onDidChangeTreeData: vscode.Event<AWSTreeNodeBase | undefined>
+    private readonly _onDidChangeTreeData: vscode.EventEmitter<AWSTreeNodeBase | undefined>
+
+    public constructor() {
+        this._onDidChangeTreeData = new vscode.EventEmitter<AWSTreeNodeBase | undefined>()
+        this.onDidChangeTreeData = this._onDidChangeTreeData.event
+    }
+
+    public initialize(context: Pick<vscode.ExtensionContext, 'asAbsolutePath' | 'globalState'>): void {
```

Comment:
`initialize` can be deleted now that the code is in activation.ts

Reply:
I wanted your guidance on that - The [AwsTreeProvider](https://github.com/aws/aws-toolkit-vscode/blob/master/src/shared/treeview/awsTreeProvider.ts#L11) interface requires it. What's the ideal path forward?

Reply:
Ah, thanks for highlighting that. I wasn't aware. Since the motivation for that was moved to the activate related ts files, we can remove initialize from the `AwsTreeProvider` interface.

Reply:
thanks, removed!